tests/unit/Makefile

```diff
@@ -59,7 +59,7 @@ $(VALGRIND_TESTS)::
 	@DYLD_LIBRARY_PATH="$(LIBCRYPTO_ROOT)/lib:$$DYLD_LIBRARY_PATH" \
 	LD_LIBRARY_PATH="$(LIBCRYPTO_ROOT)/lib:$$LD_LIBRARY_PATH" \
 	S2N_VALGRIND=1 \
-	valgrind --leak-check=full --run-libc-freeres=no -q --error-exitcode=9 --gen-suppressions=all --log-fd=2 --num-callers=40 --leak-resolution=high --undef-value-errors=no --trace-children=yes --suppressions=valgrind.suppressions \
+	valgrind --leak-check=full --show-leak-kinds=all --errors-for-leak-kinds=all --run-libc-freeres=no -q --error-exitcode=9 --gen-suppressions=all --num-callers=40 --leak-resolution=high --undef-value-errors=no --trace-children=yes --suppressions=valgrind.suppressions \
```

Comment:
Does this error if there is memory that is still reachable or does it just print something out?

Reply:
i posted local testing in the description. we now emit the error code we specify `9`

Reply:
Hmmm I suspect there will be more changes we have to make to get this to pass CI. I bet we have more "still reachable" errors that will show up in other unit tests. I guess we'll just have to wait until that PR is merged 🤷‍♀️

Reply:
> "still reachable" means your program is probably ok -- it didn't free some memory it could have. This is quite common and often reasonable.

Maybe, from what I was reading this can happen for global variables that you have not yet freed but still have a reference to. I think we should have caught most of the errors with the current CI. But yea lets see